src/main/java/org/springframework/web/servlet/DispatcherServlet.java

```diff
@@ -0,0 +1,1494 @@
+            view = mv.getView();
+            if (view == null) {
+                throw new ServletException("ModelAndView [" + mv + "] neither contains a view name nor a " +
+                        "View object in servlet with name '" + getServletName() + "'");
+            }
+        }
+
+        // Delegate to the View object for rendering.
+        if (logger.isTraceEnabled()) {
+            logger.trace("Rendering view [" + view + "] ");
+        }
+        try {
+            if (mv.getStatus() != null) {
+                response.setStatus(mv.getStatus().value());
+            }
+            view.render(mv.getModelInternal(), request, response);
+        }
+        catch (Exception ex) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Error rendering view [" + view + "]", ex);
+            }
+            throw ex;
+        }
+    }
+
+    /**
+     * Translate the supplied request into a default view name.
+     * @param request current HTTP servlet request
+     * @return the view name (or {@code null} if no default found)
+     * @throws Exception if view name translation failed
+     */
+    @Nullable
+    protected String getDefaultViewName(HttpServletRequest request) throws Exception {
+        return (this.viewNameTranslator != null ? this.viewNameTranslator.getViewName(request) : null);
+    }
+
+    /**
+     * Resolve the given view name into a View object (to be rendered).
+     * <p>The default implementations asks all ViewResolvers of this dispatcher.
+     * Can be overridden for custom resolution strategies, potentially based on
+     * specific model attributes or request parameters.
+     * @param viewName the name of the view to resolve
+     * @param model the model to be passed to the view
+     * @param locale the current locale
+     * @param request current HTTP servlet request
+     * @return the View object, or {@code null} if none found
+     * @throws Exception if the view cannot be resolved
+     * (typically in case of problems creating an actual View object)
+     * @see ViewResolver#resolveViewName
+     */
+    @Nullable
+    protected View resolveViewName(String viewName, @Nullable Map<String, Object> model,
+                                   Locale locale, HttpServletRequest request) throws Exception {
+
+        if (this.viewResolvers != null) {
+            for (ViewResolver viewResolver : this.viewResolvers) {
+                View view = viewResolver.resolveViewName(viewName, locale);
+                if (view != null) {
+                    // OSF CAS Customization: add osfUrlProperties to the model map so the view has access to OSF URLs
+                    if (this.osfUrlProperties != null ) {
+                        if (model == null) {
+                            model = new HashMap<>();
+                        }
+                        model.put("dispatcherOsfUrl", this.osfUrlProperties);
+                    }
+                    return view;
```

Comment:
For CR: this is the change, we store the `osfUrlProperties` into the `model` so that the view (i.e. FE page) will have access to it.